src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,52 @@
+package com.example.onlinebookstore.security;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final int INDEX_OF_TOKEN_START = 7;
+    private static final String TYPE_OF_TOKEN = "Bearer ";
+    private final UserDetailsService userDetailsService;
+    private final JwtUtil jwtUtil;
+
+    @Override
+    protected void doFilterInternal(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            FilterChain filterChain) throws ServletException, IOException {
+        String token = getToken(request);
+
+        if (token != null && jwtUtil.isValidToken(token)) {
+            String username = jwtUtil.getUsername(token);
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION_HEADER);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TYPE_OF_TOKEN)) {
+            return bearerToken.substring(INDEX_OF_TOKEN_START);
+        }
+        return null;
```

Comment:
you may replace it with:
```
return Optional.ofNullable(request.getHeader(HttpHeaders.AUTHORIZATION)
     .filter(header -> header.length() > TYPE_OF_TOKEN.length())
     .filter(header -> header.substring(0, TYPE_OF_TOKEN.length()).equalsIgnoreCase(TYPE_OF_TOKEN))
     .map(header -> header.substring(TYPE_OF_TOKEN.length()))
     .orElseThrow(...) // throw some custom authentication exception since something wrong
```